v-next/hardhat/src/internal/core/global-options.ts

```diff
@@ -69,6 +69,17 @@ export function buildGlobalOptionDefinitions(
         }
       }
 
+      // @ts-expect-error -- validation against js users
```

Comment:
This is a bit surprising. I'm not sure what we should do here.

On one hand, it makes sense to check this. It certainly doesn't hurt.

On the other hand, I don't think we do validation for js users anywhere else, do we? That, combined with `ts-expect-error` being something I think we should avoid unless absolutely necessary, makes me think we should just remove this check.

Not a super strong preference though, I wouldn't mind keeping it if there's a good reason.